File: Tests/beauty.cpp
#include <map>
#include <iostream>

using namespace std;


int beauty(string s){
    map<char,int>D;
    for(char i: s) 
        ++D[i];
    
    int low = s.size();
    int high = 0;
    
    for(auto it: D){
        if (it.second < low) low = it.second;
        if (it.second > high) high = it.second;
    }
    return high - low;
}

int sum_beauty(string s){
    int n = s.size();
    int r = 0;
    for (int i = 0; i<(n-1); i++){
        for (int j = i+1; j<n; j++){
            if (i<j) r = r + beauty(s.substr(i,j));
        }   
    }
    return r ;
}


int main(){


    cout << sum_beauty("aabcb") << endl;

    return 0;
}

File: Tests/beauty.py
def beauty(s):
    #Store it because is cheaper than calling twice to obtain the len of s
    n = len(s)
    D = dict()
    #Processing input
    for i in range(n):
        if s[i] in D: D[s[i]]+=1
        else: D[s[i]] = 1
    #Obtaining maximum and minimum
    low = n 
    high = 0
    for k in D.keys():
        if  D[k] < low: low = D[k]
        if D[k] > high: high = D[k]
    return high-low

def sum_beauty(s):
    s="aabcb"
    n = len(s)
    r = 0
    for i in range(n):
        for j in range(n+1):
            if i<j: r += beauty(s[i:j])
    return r

print(beauty("aababc"))

print(sum_beauty(" "))